static/js/metrics-explorer.js

```diff
@@ -575,13 +578,15 @@ async function addQueryElement() {
             await addAlertsFormulaElement();
         }
     }
-
     // Show or hide the query close icon based on the number of queries
     updateCloseIconVisibility();
-
     setupQueryElementEventListeners(queryElement);
-
     queryIndex++;
+
+    // Apply stored settings to the new query element
+    await handleQueryAndVisualize(nextQueryName, queries[nextQueryName]);
+    updateChartTheme(selectedTheme);
```

Comment:
2. Move these two functions to `addVisualizationContainer` instead of calling them in `addQueryElement`.